src/PdfReader/index.tsx

```diff
@@ -472,36 +503,71 @@ export default function usePdfReader(args: ReaderArguments): ReaderReturn {
     });
   }, [state.scale]);
 
+  /**
+   * TODO: Update to work with sub-chapter links
+   */
   const goToPage = React.useCallback(
     async (href) => {
-      const getIndexFromHref = (href: string): number => {
-        const index = manifest?.readingOrder?.findIndex((link) => {
-          return link.href === href;
-        }) as number;
-        if (index < 0) {
-          throw new Error('Cannot find resource in readingOrder');
+      if (isSinglePDF) {
+        const getIndexFromHref = (href: string): number => {
+          const index = manifest?.toc?.findIndex((link) => {
+            return link.href === href;
+          }) as number;
```

Comment:
curious as to why we have to coerce `number` type here

Reply:
`index` could be `undefined` if `manifest` or `manifest.toc` is undefined.

Reply:
got it!